nc/views.py

```diff
@@ -169,11 +179,58 @@ def contraband_hit_rate(self, request, pk=None):
         q = Q(search_type__isnull=False)
         self.query(results, group_by=("year", "driver_race", "driver_ethnicity"), filter_=q)
         response["searches"] = results.flatten()
-        # searches
+
+        # contraband
         results = GroupedData(by="year", defaults=GROUP_DEFAULTS)
         q = Q(contraband_found=True)
         self.query(results, group_by=("year", "driver_race", "driver_ethnicity"), filter_=q)
         response["contraband"] = results.flatten()
+
+        # # contraband types
+        qs = Contraband.objects.filter(stop__agency=self.get_object())
+        # # filter down by officer if supplied
+        officer = self.request.query_params.get("officer", None)
+        if officer:
+            qs = qs.filter(stop__officer_id=officer)
+        qs = qs.annotate(
+            year=ExtractYear("stop__date__year"),
+            driver_race=F("person__race"),
+            driver_ethnicity=F("person__ethnicity"),
+            contraband_type=Case(
+                When(Q(ounces__gt=0) | Q(pounds__gt=0) | Q(dosages__gt=0) | Q(grams__gt=0) | Q(kilos__gt=0), then=Value("Drugs")),
+                When(Q(pints__gt=0) | Q(gallons__gt=0), then=Value("Alcohol")),
+                When(Q(money__gt=0), then=Value("Money")),
+                When(Q(weapons__gt=0), then=Value("Weapons")),
```

Comment:
I discovered that multiple types of contraband can be reported for the same stop row:

<img width="760" alt="image" src="https://user-images.githubusercontent.com/129679/179266357-c6777686-f43d-4cec-8378-93de85691ead.png">

So I think we need to rethink our approach a bit, since we'll under report percentages now based on the precedence of the case statement (e.g. only returning Drugs when Alcohol was also found).

Reply:
Ohh! Shoot, I figured it was an either/or type of query. Ok, I'll go back to the drawing board and see what I can come up with that isn't too crazy complex. Thanks!!